test/integration/auth/mongodb_aws.test.ts

```diff
@@ -136,6 +136,47 @@ describe('MONGODB-AWS', function () {
     });
   });
 
+  context('when user supplies a credentials provider', function () {
+    let providerCount = 0;
+
+    beforeEach(function () {
+      if (!awsSdkPresent) {
+        this.skipReason = 'only relevant to AssumeRoleWithWebIdentity with SDK installed';
+        return this.skip();
+      }
+      // If we have a username the credentials have been set from the URI, options, or environment
+      // variables per the auth spec stated order.
+      if (client.options.credentials.username) {
+        this.skipReason = 'Credentials in the URI on env variables will not use custom provider.';
+        return this.skip();
+      }
+    });
+
+    it('authenticates with a user provided credentials provider', async function () {
+      // @ts-expect-error We intentionally access a protected variable.
+      const credentialProvider = AWSTemporaryCredentialProvider.awsSDK;
+      const provider = async () => {
+        providerCount++;
+        return await credentialProvider.fromNodeProviderChain().apply();
```

Comment:
Is "apply" doing something special here?

Reply:
Nothing special, just calling the function. `fromNodeProviderChain` returns a function that returns a promise of credentials. We need to actually call that function here.